From 440415703692af4548e836832ef0434e87fbc357 Mon Sep 17 00:00:00 2001
From: Oskari Lemmela <oskari@lemmela.net>
Date: Sat, 13 Jul 2024 18:56:59 +0300
Subject: [PATCH] net: ag71xx: fix qca9530 and qca9550 mdio probe

Newer QCA9530 and QCA9550 devices should use same div table as AR933X.

Fixes: d51b6ce441d3 ("net: ethernet: add ag71xx driver")
Signed-off-by: Oskari Lemmela <oskari@lemmela.net>
---
 drivers/net/ethernet/atheros/ag71xx.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/net/ethernet/atheros/ag71xx.c
+++ b/drivers/net/ethernet/atheros/ag71xx.c
@@ -641,7 +641,8 @@ static int ag71xx_mdio_get_divider(struc
 	if (!ref_clock)
 		return -EINVAL;
 
-	if (ag71xx_is(ag, AR9330) || ag71xx_is(ag, AR9340)) {
+	if (ag71xx_is(ag, AR9330) || ag71xx_is(ag, AR9340) ||
+	    ag71xx_is(ag, QCA9530) || ag71xx_is(ag, QCA9550)) {
 		table = ar933x_mdio_div_table;
 		ndivs = ARRAY_SIZE(ar933x_mdio_div_table);
 	} else if (ag71xx_is(ag, AR7240)) {
